stl/inc/memory

```diff
@@ -173,22 +290,26 @@ namespace ranges {
         }
 
     private:
-        template <class _It1, class _Se1, class _It2, class _Se2>
-        _NODISCARD static uninitialized_move_result<_It1, _It2> _Uninitialized_move_unchecked(
-            _It1 _IFirst, const _Se1 _ILast, _It2 _OFirst, const _Se2 _OLast) {
-            _STL_INTERNAL_STATIC_ASSERT(input_iterator<_It1>);
-            _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se1, _It1>);
-            _STL_INTERNAL_STATIC_ASSERT(_No_throw_forward_iterator<_It2>);
-            _STL_INTERNAL_STATIC_ASSERT(_No_throw_sentinel_for<_Se2, _It2>);
-            _STL_INTERNAL_STATIC_ASSERT(constructible_from<iter_value_t<_It2>, iter_rvalue_reference_t<_It1>>);
-
-            _Uninitialized_backout _Backout{_STD move(_OFirst)};
-
-            for (; _IFirst != _ILast && _Backout._Last != _OLast; ++_IFirst) {
-                _Backout._Emplace_back(_RANGES iter_move(_IFirst));
-            }
+        template <class _It, class _Se, class _Out, class _OSe>
+        _NODISCARD static uninitialized_move_result<_It, _Out> _Uninitialized_move_unchecked(
+            _It _IFirst, const _Se _ILast, _Out _OFirst, const _OSe _OLast) {
+            _STL_INTERNAL_STATIC_ASSERT(input_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
+            _STL_INTERNAL_STATIC_ASSERT(_No_throw_forward_iterator<_Out>);
+            _STL_INTERNAL_STATIC_ASSERT(_No_throw_sentinel_for<_OSe, _Out>);
+            _STL_INTERNAL_STATIC_ASSERT(constructible_from<iter_value_t<_Out>, iter_rvalue_reference_t<_It>>);
+
+            if constexpr (is_same_v<_Se, _It> && _Ptr_copy_cat<_It, _Out>::_Really_trivial) {

```

Comment:
This is `_Uninitialized_move_unchecked`, so I believe it should be checking `_Ptr_move_cat`?

Reply:
Absolutely Thanks for catching this